### ISSUE TITLE
Uploading a ~90GB file from stdin fails

### ISSUE THREAD

**Issue by leezu** [aws/aws-cli#6120]
When uploading a ~90GB file that is streamed from stdin, awscli fails as it selects a too small parts size and runs into a part number limit: `UploadPart operation: Part number must be an integer between 1 and 10000, inclusive` The issue occurs after a few hours, supposedly once 10000 parts have been uploaded already.

For example, the following can trigger the issue:

```
aws s3 cp s3://bucket/some-large-file.xz - | xz -d | zstd -T0 -19 | aws s3 cp - s3://bucket/some-large-file.zst
```

Please increase the part size chosen by default when copying files from stdin.

**Comment by stobrien89** [comment]
Hi @leezu,

Thanks for reaching out! Posting some of the points from our conversation here for visibility: 

Appending `--expected-size <file_size_in_bytes>` should resolve this.  The explanation of the behavior is found in the option description, but `cp` has quite a few options and you have to do a bit of digging to find this.  

> --expected-size (string) This argument specifies the expected size of a stream in terms of bytes. Note that this argument is needed only when a stream is being uploaded to s3 and the size is larger than 50GB. Failure to include this argument under these conditions may result in a failed upload due to too many parts in upload

I agree that an error like `Upload failed. When uploading files larger than 50GB, you must specify --expected-size to cp`  would be more helpful than the one currently provided and will discuss this with the team tomorrow.

**Comment by stobrien89** [comment]
Hi @leezu,

Thanks again for bringing this to our attention.  I discussed with the team and we agreed that adding an example (#6180) that shows usage of the `--expected-size` option with an explanation of how an upload may fail would bring better visibility to this and be more helpful than changing the error message, as the error still only comes after the upload has reached 50GB in size.  Closing, but let us know if you have any additional questions!
